### PR TITLE
Bug 2030988: remove log4j possibility to use JNDI

### PR DESCRIPTION
**Pull request by bentito**
Changes to opt script for CI and ART build conditions

### CODE REVIEW

File: opt_maven_install.sh [hunk 1]
```diff
@@ -36,6 +38,10 @@ else
 	  tar -xvf /tmp/hive-bin.tar.gz -C /tmp \
 	  && mv /tmp/apache-hive-${HIVE_VERSION}.redhat-${RH_HIVE_PATCH_VERSION}-bin/ \
 	  $HIVE_OUT
+	  #   remove log4j possibility to use JNDI
+	  pushd $HIVE_OUT
+      zip -q -d log4j-core-*.jar org/apache/logging/log4j/core/lookup/JndiLookup.class
+      popd
```
Comment (timflannagan) [inline opt_maven_install.sh]: Can we remove the need to perform pushd/popd operations by supplying the path to the log4j-core-related jobs explicitly? We may also need to point it to the lib directory too:

```bash
echo "Removing the log4j-core-related jars"
zip -q -d ${HIVE_OUT}/lib/log4j-core-*.jar org/apache/logging/log4j/core/lookup/JndiLookup.class

echo "Checking whether the jars were removed"
find ${HIVE_OUT} -name "log4j-core-*.jar"
```

Also: do we need to handle the edge case where zip fails and kills the container image build?

Comment (timflannagan) [reply opt_maven_install.sh]: >/tmp/opt_maven_install.sh: line 24: zip: command not found

Let's just pivot to using `find`?

Comment (bentito) [reply opt_maven_install.sh]: zip is decompressing too, added commit to add zip to yum installs on the CI portion of opt script, not sure if ART portion will need it...

File: opt_maven_install.sh [hunk 2]
```diff
@@ -8,6 +8,7 @@ export MYSQL_VERSION=8.0.21
 if [[ "$1" == "true" ]]; then
 
 	yum -y install --setopt=skip_missing_names_on_install=False \
+		zip \
```
Comment (timflannagan) [inline opt_maven_install.sh]: I think we'll need to also install zip for the ART portion too, so we may want to update the yum install command here: https://github.com/kube-reporting/hive/blob/master/Dockerfile.rhel8#L3.

Comment (bentito) [reply opt_maven_install.sh]: roger

File: Dockerfile.rhel8 [hunk 3]
```diff
@@ -30,6 +30,9 @@ WORKDIR $HIVE_HOME
 
 ENV HADOOP_CLASSPATH $HIVE_HOME/hcatalog/share/hcatalog/*:${HADOOP_CLASSPATH}
 
+# make JNDI impossible from log4j usages
+RUN zip -q -d lib/log4j-core-*.jar org/apache/logging/log4j/core/lookup/JndiLookup.class
```
Comment (timflannagan) [inline Dockerfile.rhel8:34]: Just a quick note for our future selves: https://github.com/kube-reporting/hive/pull/68 exposed that the hadoop base image doesn't have zip present.

**Comment by timflannagan** [comment]
It looks like our tide configuration for this organization/repository is stale since the removal of this component in 4.9. I'm going to manually label the PR as Peter/QE already ACKed these changes.

**Pull request merged**